rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java

```diff
@@ -88,9 +87,10 @@ protected T executeAutoCommitTrue(Object[] args) throws Throwable {
         LockRetryController lockRetryController = new LockRetryController();
         try {
             connectionProxy.setAutoCommit(false);
+
+            result = executeAutoCommitFalse(args);
```

Comment:
During the retry, the before-image data may change. This is wrong.

Reply:
1. prepareUndoLog() should clear undolog in this case.
2. before-image cost much time in batch-situation. 

I will remove this changed line, discuss it in other issue.

Reply:
1. yeah, undolog should add in Set but List.
2. batch solution should use autocommit=false